system/Helpers/form_helper.php

```diff
@@ -0,0 +1,954 @@
+
+if ( ! function_exists('form_keygen'))
+{
+    function form_keygen()
+    {
+        /* The purpose of the <keygen> element is to provide a secure way to authenticate users.
+         The <keygen> element specifies a key-pair generator field in a form.
+         When the form is submitted, two keys are generated, one private and one public.
+         The private key is stored locally, and the public key is sent to the server.
+         The public key could be used to generate a client certificate to authenticate the user in the future. */
+    }
+}
+
+// ------------------------------------------------------------------------
+
+if ( ! function_exists('form_output'))
+{
+    function form_output()
+    {
+        /* The <output> element represents the result of a calculation (like one performed by a script). */
+    }
+}
+
+// ------------------------------------------------------------------------
+
+if ( ! function_exists('form_fieldset'))
+{
+    /**
+     * Fieldset Tag
+     *
+     * Used to produce <fieldset><legend>text</legend>.  To close fieldset
+     * use form_fieldset_close()
+     *
+     * @param	string	The legend text
+     * @param	array	Additional attributes
+     * @return	string
+     */
+    function form_fieldset(string $legend_text = '', array $attributes = []): string
+    {
+        $fieldset = '<fieldset'._attributes_to_string($attributes).">\n";
+        
+        if ($legend_text !== '')
+        {
+            return $fieldset.'<legend>'.$legend_text."</legend>\n";
+        }
+        
+        return $fieldset;
+    }
+}
+
+// ------------------------------------------------------------------------
+
+if ( ! function_exists('form_fieldset_close'))
+{
+    /**
+     * Fieldset Close Tag
+     *
+     * @param	string
+     * @return	string
+     */
+    function form_fieldset_close(string $extra = ''): string
+    {
+        return '</fieldset>'.$extra;
+    }
+}
+
+// ------------------------------------------------------------------------
+
+if ( ! function_exists('form_close'))
+{
+    /**
+     * Form Close Tag
+     *
+     * @param	string
+     * @return	string
+     */
+    function form_close(string $extra = ''): string
+    {
+        return '</form>'.$extra;
+    }
+}
+
+// ------------------------------------------------------------------------
+
+if ( ! function_exists('set_value'))
+{
+    /**
+     * Form Value
+     *
+     * Grabs a value from the POST array for the specified field so you can
+     * re-populate an input field or textarea
+     *
+     * @param	string	$field		Field name
+     * @param	string	$default	Default value
+     * @param	bool	$html_escape	Whether to escape HTML special characters or not
+     * @return	string
+     */
+    function set_value(string $field, string $default = '', bool $html_escape = true): string
+    {
+        $value = $_POST[$field] ?? $default;
+        
+        return ($html_escape) ? html_escape($value) : $value;
```

Comment:
No function named `html_escape` exists in the framework. You should be using the `esc()` method instead. Which will also require you ensure it's loaded....